app/controllers/restaurants_controller.rb

```diff
@@ -2,7 +2,12 @@ class RestaurantsController < ApplicationController
   before_action :authorize_user, except: [:index, :show, :create, :new]
 
   def index
-    @restaurants = Restaurant.order("name ASC").page(params[:page])
+    if params[:search_text].present?
+      @restaurants =
+        Restaurant.search_results(params[:search_text]).page(params[:page])
```

Comment:
the method name `search_results` is a bit unintuitive. I would consider renaming it to just `search`